src/dvc_data/index/diff.py

```diff
@@ -102,13 +108,20 @@ def _diff_entry(
     if hash_only:
         return hi_diff
 
-    if meta_diff != UNCHANGED:
+    # If both meta's are None, return hi_diff
+    if meta_diff == UNCHANGED and old_meta is None:
+        return hi_diff
+
+    # If both hi's are falsey, return meta_diff
+    if hi_diff == UNCHANGED and not old_hi:
```

Comment:
I'm wondering if it's actually safe for us to rely on `bool(HashInfo)` in `diff` vs using `is None` comparisons instead. We get a false-y hashinfo any time `hashinfo.value` is false-y, but it seems like we should be accounting for the potential difference between `HashInfo(name="md5", value="")` and `HashInfo(name="etag", value="")`